src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -729,5 +750,47 @@ public void RaiseNotificationForBatchEnded(string subscriptionName, Subscription
         {
             OnEndBatch?.Invoke(subscriptionName, batchAggregator);
         }
+
+        public static SubscriptionGeneralDataAndStats GetSubscriptionFromServerStore(ServerStore serverStore, TransactionOperationContext context, string databaseName, string name)
+        {
+            var subscriptionBlittable = serverStore.Cluster.Read(context, SubscriptionState.GenerateSubscriptionItemKeyName(databaseName, name));
+
+            if (subscriptionBlittable == null)
+                throw new SubscriptionDoesNotExistException($"Subscription with name '{name}' was not found in server store");
+
+            var subscriptionState = JsonDeserializationClient.SubscriptionState(subscriptionBlittable);
+            var subscriptionJsonValue = new SubscriptionGeneralDataAndStats(subscriptionState);
+            return subscriptionJsonValue;
+        }
+
+        public static SubscriptionState GetSubscriptionFromServerStoreById(ServerStore serverStore, string databaseName, long id)
+        {
+            using (serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverStoreContext))
+            using (serverStoreContext.OpenReadTransaction())
+            {
+                var name = GetSubscriptionNameById(serverStoreContext, databaseName, id);
+                if (string.IsNullOrEmpty(name))
+                    throw new SubscriptionDoesNotExistException($"Subscription with id '{id}' was not found in server store");
+
+                return GetSubscriptionFromServerStore(serverStore, serverStoreContext, databaseName, name);
+            }
+        }
+
+        public static string GetSubscriptionNameById(TransactionOperationContext serverStoreContext, string databaseName, long id)
```

Comment:
can we introduce serverStore.Cluster.Subscriptions.Read & friends?

Reply:
done